packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts

```diff
@@ -496,12 +496,11 @@ export class DotNetGenerator extends Generator {
         overrides = true;
       }
     }
-    if (method.static) {
-      staticKeyWord = 'static ';
-    } else if (overrides) {
+    if (overrides) {
       // Add the override key word if the method is emitted for a proxy or data type or is defined on an ancestor
-      overrideKeyWord = 'override ';
+      overrideKeyWord = method.static ? 'new ' : 'override ';
     } else if (
+      !method.static &&
```

Comment:
I have been internally debating this a little bit... Initially I wanted to simply change the compiler so that statics are **never** considered to be overrides; but this caused problems, because in JavaScript (and hence, TypeScript), they actually are overrides... This manifests when you try the following:

```ts
export class Parent { public static foo: number; }

//! Child.foo incorrectly implements Parent.foo (types don't match)
export class Child extends Parent { public static foo: string; }
```

This made me decide that it was a more coherent story from the author's standpoint if we reproduced the ES6 semantics in the jsii assembly (where statics are inherited with the rest of the prototype). Carrying them as overrides has advantages - in languages where static are inherited (although JS is the only one that comes to mind now), this allows overrides to use `super`.

We *could* distinguish static overrides from non-static overrides, however this distinction is very much a code generation concern, and not so much a type model concern (after all, all we care about is whether this is an override or not, period).